nft-rfc-006.md

```diff
@@ -0,0 +1,860 @@
+chain agnostic identifiers must allow for future chains to use the same
+approach, without changing the underlying IID specification.
+
+Requirement 4 -- Enable verifiable assertions
+---------------------------------------------
+
+The identifiers for both on-chain assets and off-chain assets must be
+suitable for verifiable assertions to be made about both, without loss
+of rigor.
+
+Requirement 5 -- Enable both private and public assertions
+----------------------------------------------------------
+
+The IID approach must enable both publicly revealed assertions, which
+are available to anyone without further authentication or permission,
+and private assertions which must be attained through privacy-respecting
+means.
+
+Requirement 6 -- Verifiability of information completeness
+----------------------------------------------------------
+
+Potential buyers of NFTs must be able to prove they have all of the
+relevant information for the use of the NFT---at least as asserted by
+the current token record. In order to prevent privacy problems, this
+information may be private---requiring the buyer to secure the
+information through permissioned means---but it must be possible for a
+buyer to know if they have appropriately acquired all of those details.
+For example, a title property may need to disclose a lien against the
+property, without revealing publicly the details of that lien, because
+the lien itself may expose private information. Knowing that some form
+of additional data is required allows a serious buyer to request that
+information without leaking private data to the public. Observers must
+be able to cryptographically verify that such additional data \*is\* the
+data associated with the NFT and that all such data has been collected.
+
+Requirement 7 -- Off-chain creation of cryptographic identifiers
+----------------------------------------------------------------
+
+When supportable by a given chain, it must be possible for a new
+identifier to be minted and usable before attaching it to an onchain
+token. These latent tokens may be used for demonstrating
+proof-of-control, authentication based on proof-of-control, and
+cryptographic signing, without incurring the transactional costs and
+delays of interacting with the foundational chain. Updates and
+annotations to identifier related metadata would require chain
+interaction, but the initial minting would not. This feature allows the
+use of such identifiers for assertions, such as verifiable credentials,
+prior to anchoring to a chain. The verification for such an identifier
+must check on-chain for updates, but finding none, the identifier itself
+can be used to deterministically generate the appropriate metadata for
+its use.
+
+Requirement 8 -- Use with self-sovereign identity
+-------------------------------------------------
+
+IIDs must be able to work with emerging self-sovereign identity
+approaches, which allows individuals to manage identifiers and
```

Comment:
Ideally, this can be extended beyond just supporting individuals. Of course, multisigs get us partway there by allowing a well-defined mostly constant set of individuals to identify themselves.

However, a DID module in the interchain ecosystem would greatly benefit from also allowing more complex entities (like a validator set, or blockchain governance) identify themselves and additionally specify how they will authenticate for **on-chain** services (like IBC applications).

This would allow all applications to target DIDs as a mode for authenticating users, and automatically support a diverse set of entities and authentication schemes interacting with their application.

I've written down the design I have in mind here: https://github.com/cosmos/cosmos-sdk/discussions/8431

Of course this extensibility has privacy tradeoffs. I wonder if there's a way to build the design I have in mind with the current IID work, and if not what might be necessary to achieve it.

Reply:
Absolutely. Authentication wrt token control or use should be a straightforward application of existing DID Auth approaches. As such, the auth could be performed by anyone who already expects IIDs to act like DIDs, whether that is a person or a machine or an organization.

In addition, using IID References, we can specify in the IID document how any number of parties (human or otherwise) might authenticate cryptographically for interactions related to the token, without conflating the authenticating party with the token itself.

For example, the owner of an IID wedding invitation token would have their ownership rights managed by DID Auth, while, in the same IID Document, a "plusOne" can authenticate themselves using different cryptography.

This approach should be more fully described. In particular, we probably should add to "Associated Resources" this resource type "Reference", where #plusOne would be defined with multiple ways to make statements about #plusOne, such as their privilege of attendance, plus authentication methods and verification relationships, either embedded or externalized. 

The most privacy respecting option would be to have a single external resource entry that lists a hashgraph. In the #plusOne use case, the person claiming "plusOne" status would show up with a Auth specification and merkle filter that proves their auth mechanism was included as an associated resource with that particular benefit. Under the hood, this just looks like a Digital Invitation. 

Alternatively, these features could be enabled by delegatable capabilities, which might have some better features than modeling them as statements in JSON-LD. That, of course, requires selecting a capabilities architecture, but such capabilities should be easily included as Associated Resources without loss.